src/Cli/dotnet/commands/dotnet-sln/add/Program.cs

```diff
@@ -102,6 +102,12 @@ private async Task AddProjectsToSolutionAsync(string solutionFileFullPath, IEnum
                 string relativePath = Path.GetRelativePath(Path.GetDirectoryName(solutionFileFullPath), projectPath);
                 // Add fallback solution folder
                 string relativeSolutionFolder = Path.GetDirectoryName(relativePath);
+                // Don't add solution folder if it contains ..
+                if (relativeSolutionFolder.Contains(".."))
+                {
+                    relativeSolutionFolder = string.Empty;
+                }
+
```

Comment:
Another solution might be stripping the .. and adding the subfolders accordingly, allthough I believe that could end up being confusing, but am happy to discuss it :)